azurerm/internal/services/machinelearning/tests/resource_arm_machine_learning_workspace_test.go

```diff
@@ -0,0 +1,361 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMMachineLearningWorkspace_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMachineLearningWorkspaceExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMMachineLearningWorkspace_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMachineLearningWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMMachineLearningWorkspace_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMachineLearningWorkspaceExists(data.ResourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMMachineLearningWorkspace_requiresImport(data),
+				ExpectError: acceptance.RequiresImportError("azurerm_machine_learning_workspace"),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMMachineLearningWorkspace_withTags(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMachineLearningWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMMachineLearningWorkspaceWithTags(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMachineLearningWorkspaceExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "2"),
+				),
+			},
+			{
+				Config: testAccAzureRMMachineLearningWorkspace_withTagsUpdate(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMachineLearningWorkspaceExists(data.ResourceName),
+					resource.TestCheckResourceAttr(data.ResourceName, "tags.%", "1"),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMMachineLearningWorkspace_withContainerRegistry(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_machine_learning_workspace", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMMachineLearningWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMMachineLearningWorkspace_withContainerRegistry(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMMachineLearningWorkspaceExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMMachineLearningWorkspace_complete(t *testing.T) {
```

Comment:
Could we have an update test where we go basic -> complete -> basic with an import step after each

Reply:
Well, this cannot be done in this resource, because the complete test has an extra `container_registry_id` set, which is `forceNew: true`, therefore I think this resource does not need an update test by basic -> complete -> basic.
The things that can be updated is the `tags`, `description`, etc. I can do a update test on these attributes.